libraries/botframework-connector/src/auth/botFrameworkAuthentication.ts

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { Activity, CallerIdConstants } from 'botframework-schema';
+import { AuthenticateRequestResult } from './authenticateRequestResult';
+import { BotFrameworkClient } from '../skills';
+import { ClaimsIdentity } from './claimsIdentity';
+import { ConnectorClient } from '../connectorApi/connectorClient';
+import { JwtTokenValidation } from './jwtTokenValidation';
+import { ServiceClientCredentialsFactory } from '../serviceClientCredentialsFactory';
+import { SkillValidation } from './skillValidation';
+import { UserTokenClient } from './userTokenClient';
+
+/**
+ * Represents a Cloud Environment used to authenticate Bot Framework Protocol network calls within this environment.
+ */
+export abstract class BotFrameworkAuthentication {
+    /**
+     * Validate Bot Framework Protocol requests.
+     *
+     * @param activity The inbound Activity.
+     * @param authHeader The HTTP auth header.
+     */
+    abstract authenticateRequest(activity: Activity, authHeader: string): Promise<AuthenticateRequestResult>;
+
+    /**
+     * Validate Bot Framework Protocol requests.
+     *
+     * @param authHeader The HTTP auth header.
+     * @param channelIdHeader The channel ID HTTP header.
+     */
+    abstract authenticateStreamingRequest(
+        authHeader: string,
+        channelIdHeader: string
+    ): Promise<AuthenticateRequestResult>;
+
+    /**
+     * Creates a ConnectorFactory that can be used to create ConnectorClients that can use credentials from this particular Cloud Environment.
+     *
+     * @param claimsIdentity The inbound Activity's ClaimsIdentity.
+     */
+    abstract createConnectorFactory(claimsIdentity: ClaimsIdentity): ConnectorClient;
+
+    /**
+     * Creates the appropriate UserTokenClient instance.
+     *
+     * @param claimsIdentity The inbound Activity's ClaimsIdentity.
+     */
+    abstract createUserTokenClient(claimsIdentity: ClaimsIdentity): Promise<UserTokenClient>;
+
+    /**
+     * Creates a BotFrameworkClient for calling Skills.
+     */
+    createBotFrameworkClient(): BotFrameworkClient {
+        return notImplementedError();
+    }
+
+    /**
+     * Gets the originating audience from Bot OAuth scope.
+     */
+    getOriginatingAudience(): string {
+        return notImplementedError();
+    }
+
+    // TODO: Update docstring - this is a direct port from .NET
+    /**
+     * Authenticate Bot Framework Protocol request to Skills.
+     *
+     * @param authHeader The HTTP auth header in the skill request.
+     */
+    authenticateChannelRequest(authHeader: string): Promise<ClaimsIdentity> {
```

Comment:
`authenticateChannelServiceRequest` might have been more accurate, albeit tied to a concept from different layer; [`botbuilder-core`](https://github.com/microsoft/botbuilder-js/blob/6bd641eb13dd4e09dc4bb4fac4b3155f3005922f/libraries/botbuilder/src/channelServiceHandler.ts#L37)/[`Microsoft.Bot.Builder`](https://github.com/microsoft/botbuilder-dotnet/blob/03110e331220592f142bee4d58bdbb395051403e/libraries/Microsoft.Bot.Builder/ChannelServiceHandlerBase.cs#L16) and [`botbuilder`](https://github.com/microsoft/botbuilder-js/blob/6bd641eb13dd4e09dc4bb4fac4b3155f3005922f/libraries/botbuilder/src/channelServiceRoutes.ts#L38)/[`Microsoft.Bot.Builder.Integration.AspNet.Core`](https://github.com/microsoft/botbuilder-dotnet/blob/03110e331220592f142bee4d58bdbb395051403e/libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/ChannelServiceController.cs#L15)